<?xml version="1.0"?>
<launch>

    <!-- Parametrized sample launch file to run pybullet ros wrapper -->

    <!-- relevant configuration for pybullet_ros node -->
    <arg name="loop_rate" default="80.0" />
    <arg name="pybullet_gui" default="True" />
    <arg name="robot_urdf_path" default="$(find pybullet_ros)/common/test/urdf/acrobat_robot.urdf" />
    <arg name="pause_simulation" default="False" />
    <arg name="gravity" default="-9.81" />
    <arg name="max_effort_vel_mode" default="50.0" />
    <arg name="robot_pose_x" default="0.0" />
    <arg name="robot_pose_y" default="0.0" />
    <arg name="robot_pose_z" default="1.0" />
    <arg name="robot_pose_yaw" default="0.0" />
    <arg name="fixed_base" default="True" />

    <!-- pybullet ros wrapper node -->
    <node pkg="pybullet_ros" type="pybullet_ros_node" name="pybullet_ros" output="screen" required="true" >
        <param name="loop_rate" value="$(arg loop_rate)" />
        <param name="pybullet_gui" value="$(arg pybullet_gui)" />
        <param name="robot_urdf_path" value="$(arg robot_urdf_path)" />
        <param name="pause_simulation" value="$(arg pause_simulation)" />
        <param name="gravity" value="$(arg gravity)" />
        <param name="max_effort_vel_mode" value="$(arg max_effort_vel_mode)" />
        <param name="robot_pose_x" value="$(arg robot_pose_x)" />
        <param name="robot_pose_y" value="$(arg robot_pose_y)" />
        <param name="robot_pose_z" value="$(arg robot_pose_z)" />
        <param name="robot_pose_yaw" value="$(arg robot_pose_yaw)" />
        <param name="fixed_base" value="$(arg fixed_base)" />
    </node>

</launch>
